node/core/backing/src/lib.rs

```diff
@@ -797,9 +797,15 @@ async fn handle_active_leaves_update<Context>(
 	// as a result.
 	//
 	// when prospective parachains are disabled, the implicit view is empty,
-	// which means we'll clean up everything. This is correct.
```

Comment:
This was not correct. The old behavior was to clean up everything except the leaf that was just activated, when prospective parachains weren't active. The new behavior is to make sure every active leaf is preserved.